File: src/java/cyr7/ast/expr/ExprNode.java
package cyr7.ast.expr;

import cyr7.ast.Node;
import cyr7.semantics.types.ExpandedType;

/**
 * Represents a generic expression: Ex: 1+1, true, arr[2][3]
 */
public interface ExprNode extends Node {

    ExpandedType getType();

    ExprNode setType(ExpandedType t);

}

File: src/java/cyr7/ast/expr/ExprNode.kt
package cyr7.ast.expr

import cyr7.ast.Node
import cyr7.semantics.types.ExpandedType

/**
 * Represents a generic expression: Ex: 1+1, true, arr[2][3]
 */
interface ExprNode : Node {
    val type: ExpandedType?
    fun setType(t: ExpandedType): ExprNode
}
